configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java

```diff
@@ -281,6 +282,27 @@ public int deleteExpiredRemoteSessions(Clock clock, Duration expiryTime) {
         return deleted;
     }
 
+    public int deleteExpiredLocks(Clock clock, Duration expiryTime) {
+        int deleted = 0;
+        for (var lock : curator.getChildren(locksPath)) {
+            Path path = locksPath.append(lock);
+            if (zooKeeperNodeCreated(path).isBefore(clock.instant().minus(expiryTime))) {
+                log.log(Level.INFO, "Lock  " + path + " has expired, deleting it");
+                curator.delete(path);
+                deleted++;
+            }
+        }
+        return deleted;
+    }
+
+    private Instant zooKeeperNodeCreated(Path path) {
```

Comment:
perhaps this should return an Optional instead, and empty where it now returns the current time?

Reply:
Yes, probably better, thanks